rest_framework/serializers.py

```diff
@@ -440,7 +443,9 @@ def to_internal_value(self, data):
             except ValidationError as exc:
                 errors[field.field_name] = exc.detail
             except DjangoValidationError as exc:
-                errors[field.field_name] = list(exc.messages)
+                errors[field.field_name] = (
+                    exceptions.build_error_from_django_validation_error(exc.messages)
```

Comment:
Nice catch. Looks like I'm missing a test case that covers this path. I'll cover this and push an update. Hopefully tomorrow :)